examples/FreeEnergy.cpp

```diff
@@ -0,0 +1,72 @@
+#include <Cabana_Grid.hpp>
+#include <PFHub.hpp>
+#include <cassert>
+
+using namespace CabanaPF;
+using ptr = std::unique_ptr<PFHub1aBase>;
+
+// settings for how problems will be run:
+const int GRID_POINTS_1D = 256;
+const double DT = .1;
+
+/*
+    Runs each simulation passed in through the problems array
+    Prints the free energy and outputs the c grid every 5 seconds up to 100, then every 50 up to 1000, and so on until
+   end_time
+*/
+void print_free_energy_csv(std::vector<ptr>& problems, int end_time) {
+    assert(end_time >= 10);
+
+    // print names and title line:
+    for (const auto& problem_ptr : problems)
+        std::cout << "," << problem_ptr->subproblem_name();
+    std::cout << "\ntime";
+    for ([[maybe_unused]] const auto& problem_ptr : problems)
+        std::cout << ",free_energy";
+    std::cout << std::endl;
+
+    // Ad-hoc calculate at t=0, 5, and 10:
+    for (int t = 0; t <= 10; t += 5) {
+        std::cout << t;
+        for (const auto& problem_ptr : problems) {
+            problem_ptr->run_until_time(t);
+            problem_ptr->snapshot_c();
+            std::cout << "," << problem_ptr->free_energy();
+        }
+        std::cout << std::endl;
+    }
+    // Loop that calculates the energy and saves a c grid at (15, 20, ... 100), (150, 200, ... 1000), etc
+    int output_every = 5;
+    while (round(problems[0]->get_time_done()) + output_every <= end_time) {
+        // sanity check:
+        for (const auto& problem_ptr : problems)
+            assert(round(problem_ptr->get_time_done()) == 2 * output_every);
+        // outputs 18 lines, or fewer if user wants to stop partway through an order of magnitude:
+        for (int t = 3 * output_every; t <= std::min(20 * output_every, end_time); t += output_every) {
+            std::cout << t;
+            for (const auto& problem_ptr : problems) {
+                problem_ptr->run_until_time(t);
+                problem_ptr->snapshot_c();
+                std::cout << "," << problem_ptr->free_energy();
+            }
+            std::cout << std::endl;
+        }
+        output_every *= 10;
+    }
+}
+
+int main(int argc, char* argv[]) {
```

Comment:
I think we want to call this file/exe something else - it's more of a comparison of all the variants we've set up so far